flash/image/face_detection/model.py

```diff
@@ -0,0 +1,238 @@
+        learning_rate: The learning rate to use for training
+
+    """
+
+    required_extras: str = "image"
+
+    def __init__(
+        self,
+        model: str = "lffd_slim",
+        pretrained: bool = True,
+        loss=None,
+        metrics: Union[Callable, nn.Module, Mapping, Sequence, None] = None,
+        optimizer: Type[Optimizer] = torch.optim.AdamW,
+        learning_rate: float = 1e-4,
+        serializer: Optional[Union[Serializer, Mapping[str, Serializer]]] = None,
+        preprocess: Optional[Preprocess] = None,
+        **kwargs: Any,
+    ):
+        self.save_hyperparameters()
+
+        if model in ff.list_pretrained_models():
+            model = FaceDetector.get_model(model, pretrained, **kwargs)
+        else:
+            ValueError(f"{model} is not supported yet.")
+
+        super().__init__(
+            model=model,
+            loss_fn=loss,
+            metrics=metrics or {"AP": ff.metric.AveragePrecision()},
+            learning_rate=learning_rate,
+            optimizer=optimizer,
+            serializer=serializer or DetectionLabels(),
+            preprocess=preprocess or FaceDetectionPreprocess(),
+        )
+
+    @staticmethod
+    def get_model(
+        model_name,
+        pretrained,
+        **kwargs,
+    ):
+
+        if pretrained:
+            pl_model = ff.FaceDetector.from_pretrained(model_name, **kwargs)
+        else:
+            arch, config = model_name.split("_")
+            pl_model = ff.FaceDetector.build(arch, config, **kwargs)
+
+        # get torch.nn.Module
+        model = getattr(pl_model, "arch")
+
+        # set preprocess params
+        model.register_buffer("normalizer", getattr(pl_model, "normalizer"))
+        model.register_buffer("mean", getattr(pl_model, "mean"))
+        model.register_buffer("std", getattr(pl_model, "std"))
+
+        # set postprocess function
+        setattr(model, "_postprocess", getattr(pl_model, "_postprocess"))
+
+        return model
+
+    def forward(self, x: List[torch.Tensor]) -> Any:
+
+        batch, scales, paddings = ff.utils.preprocess.prepare_batch(x, None, adaptive_batch=True)
+        # batch: torch.Tensor(B,C,T,T)
+        # scales: torch.Tensor(B,)
+        # paddings: torch.Tensor(B,4) as pad (left, top, right, bottom)
+
+        # apply preprocess
+        batch = (((batch * 255) / self.model.normalizer) - self.model.mean) / self.model.std
+
+        # get logits
+        logits = self.model(batch)
+        # logits, any
+
+        preds = self.model.logits_to_preds(logits)
+        # preds: torch.Tensor(B, N, 5)
+
+        preds = self.model._postprocess(preds)
+        # preds: torch.Tensor(N, 6) as x1,y1,x2,y2,score,batch_idx
+
+        preds = [preds[preds[:, 5] == batch_idx, :5] for batch_idx in range(batch.size(0))]
+        # preds: list of torch.Tensor(N, 5) as x1,y1,x2,y2,score
+
+        preds = ff.utils.preprocess.adjust_results(preds, scales, paddings)
```

Comment:
This can be done in a postprocess, retrieving scales and paddings from the metadata